### PR TITLE
azure: Fix CCM deployment and ssh reconnects

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:
While testing #2611 it was discovered that:
* azure CCM fails with "unknown --port flag", removing it fixes external CCM
* Flatcar control plane hosts like (for update) to reboot right in the middle of cluster initialization, which broke SSH client and left it in the "half-broken" state

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind bug

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
* azure CCM: Fix unknown port flag in deployment
* fix ssh client to re-establish broken ssh connections
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: examples/terraform/azure/variables.tf [hunk 1]
```diff
@@ -270,3 +270,9 @@ variable "rhsm_offline_token" {
   type        = string
   sensitive   = true
 }
+
+variable "disable_auto_update" {
+  description = "Flatcar specific"
```
Comment (xmudrii) [inline examples/terraform/azure/variables.tf:275]: It's obvious from the name, but let's add a proper description so it renders well in docs.

Comment (kron4eg) [reply examples/terraform/azure/variables.tf]: Done

**Pull request merged**